CONTRIBUTING.md

```diff
@@ -19,78 +19,117 @@ If you're new to open source in general, check out [GitHub's open source intro g
 All command-line operations for building and testing OpenSeadragon are scripted using [Grunt](https://gruntjs.com/) which is based on [Node.js](https://nodejs.org/). To get set up:
 
 1. Install Node, if you haven't already (available at the link above)
-1. Install the Grunt command line runner (if you haven't already); on the command line, run `npm install -g grunt-cli`
-1. Clone the openseadragon repository
-1. On the command line, go in to the openseadragon folder
-1. Run `npm install`
+2. Install the Grunt command line runner (if you haven't already); on the command line, run `npm install -g grunt-cli`
+3. Clone the openseadragon repository
+4. On the command line, go in to the openseadragon folder
+5. Run `npm install`
 
 You're set, all development dependencies should have been installed and the project built...
 continue reading for build and test instructions.
 
+### Project Structure
+
+The OpenSeadragon codebase is organized as follows:
+
+- `src/` - Source code for the core library
+- `build/` - Output directory for builds (not checked into git)
+- `test/` - Test files and demos
+- `doc/` - API documentation
+- `images/` - Images used by the demos and examples
+
+Understanding this structure will help you locate the appropriate files for your contributions.
+
 ### Building from Source
 
 To build, just run (on the command line, in the openseadragon folder):
 
-    grunt
+  - `grunt`
 
 If you want Grunt to watch your source files and rebuild every time you change one, use:
 
-    grunt watch
+   - `grunt watch`
 
 To have it watch your source files and also run a server for you to test in:
 
-    grunt dev
+   - `grunt dev`
 
 The built files appear in the `build` folder.
 
-If you want to build tar and zip files for distribution (they will also appear in the `build` folder), use:
+### Testing
 
-    grunt package
+Our tests are based on [QUnit](https://qunitjs.com/) and [Puppeteer](https://github.com/GoogleChrome/puppeteer); they're both installed when you run `npm install`. 
 
-Note that the `build` folder is masked with .gitignore; it's just for your local use, and won't be checked in to the repository.
+You can run tests using either Grunt or npm commands:
 
-You can also publish the built version to the site-build repository. This assumes you have cloned it next to this repository. The command is:
+| Task | Grunt Command | npm Command |
+|------|--------------|-------------|
+| Run all tests | `grunt test` | `npm run test` |
+| Test specific module | `grunt test --module="navigator"` | `npm run test -- --module="navigator"` |
+| Interactive testing | `grunt connect watch` | `npm run dev` |
```

Comment:
Let's actually change `grunt connect watch` to `grunt dev` which already exists and is a better choice.